src/AssetRegistry/StandardERC4626SubRegistry.sol

```diff
@@ -0,0 +1,172 @@
+            assetToInformation[asset].underlyingAssetAddress,
+            assetToInformation[asset].underlyingAssetOracleAddresses
+        );
+    }
+
+    /**
+     * @notice Checks for a token address and the corresponding Id if it is white-listed
+     * @param assetAddress The address of the asset
+     * @dev Since ERC4626 tokens have no Id, the Id should be set to 0
+     * @return A boolean, indicating if the asset passed as input is whitelisted
+     */
+    function isWhiteListed(address assetAddress, uint256)
+        external
+        view
+        override
+        returns (bool)
+    {
+        if (isAssetAddressWhiteListed[assetAddress]) {
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * @notice Returns the value of a certain asset, denominated in USD or in another BaseCurrency
+     * @param getValueInput A Struct with all the information neccessary to get the value of an asset
+     *                      - assetAddress: The contract address of the asset
+     *                      - assetId: Since ERC4626 tokens have no Id, the Id should be set to 0
+     *                      - assetAmount: The Amount of Shares, ERC4626 tokens can have any Decimals precision smaller than 18.
+     *                      - baseCurrency: The BaseCurrency (base-asset) in which the value is ideally expressed
+     * @return valueInUsd The value of the asset denominated in USD with 18 Decimals precision
+     * @return valueInBaseCurrency The value of the asset denominated in BaseCurrency different from USD with 18 Decimals precision
+     * @dev If the Oracle-Hub returns the rate in a baseCurrency different from USD, the StandardERC4626Registry will return
+     *      the value of the asset in the same BaseCurrency. If the Oracle-Hub returns the rate in USD, the StandardERC4626Registry
+     *      will return the value of the asset in USD.
+     *      Only one of the two values can be different from 0.
+     * @dev Function will overflow when assetAmount * Rate * 10**(18 - rateDecimals) > MAXUINT256
+     * @dev If the asset is not first added to subregistry this function will return value 0 without throwing an error.
+     *      However no check in StandardERC4626Registry is necessary, since the check if the asset is whitelisted (and hence added to subregistry)
+     *      is already done in the Main-Registry.
+     */
+    function getValue(GetValueInput memory getValueInput)
+        public
+        view
+        override
+        returns (uint256 valueInUsd, uint256 valueInBaseCurrency)
+    {
+        uint256 rateInUsd;
+        uint256 rateInBaseCurrency;
+
+        (rateInUsd, rateInBaseCurrency) = IOraclesHub(oracleHub).getRate(
+            assetToInformation[getValueInput.assetAddress].underlyingAssetOracleAddresses,
+            getValueInput.baseCurrency
+        );
+
+        uint256 assetAmount = IERC4626(getValueInput.assetAddress).convertToAssets(getValueInput.assetAmount);
```

Comment:
Good for now, when we optimize for gas usage, we might want to avoid this call.
We now do a .getBalance() in the vault, to get the shares. And here we jump a second time to the ERC4626 contract to convert the shares to assets.
It's a trade-off between gas usage and keeping the vault generic (don't make a difference between logic to handle ERC20 and ERC4626). Maybe create an issue that we don't forget?

Reply:
#111